support/Environments/caltech_hpc_gcc.sh

```diff
@@ -79,7 +81,7 @@ spectre_run_cmake() {
           -D CMAKE_Fortran_COMPILER=gfortran \
           -D CHARM_ROOT=$CHARM_ROOT \
           -D CMAKE_BUILD_TYPE=Release \
-          -D MEMORY_ALLOCATOR=SYSTEM \
+          -D MEMORY_ALLOCATOR=JEMALLOC \
```

Comment:
As mentioned in the call today: please add a comment that this is important and why. Maybe above the `cmake` command here.